xapian-letor/ranker/ranker.cc

```diff
@@ -114,9 +114,9 @@ static int
 getlabel(const Document & doc, const std::string & qid)
 {
     int label = -1;
-    string id = std::to_string(doc.get_docid());
-    map<string, map<string, int> >::const_iterator outerit;
-    map<string, int>::const_iterator innerit;
+    int id = doc.get_docid();
+    map<string, map<int, int> >::const_iterator outerit;
+    map<int, int>::const_iterator innerit;
 
     outerit = qrel.find(qid);
```

Comment:
It's better to declare a variable at the point where you can initialise if that's easy to do (which it is here for `outerit` and `innerit`).

Also with C++11 better in a case like this to just use `auto` rather than manually writing out the long templated typenames.